msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java

```diff
@@ -781,12 +768,71 @@ public void acquireTokenSilentAsync(@NonNull final String[] scopes, @NonNull fin
         acquireTokenSilent(scopes, user, authority, forceRefresh, wrapCallbackForTelemetryIntercept(apiEventBuilder, callback), telemetryRequestId, apiEventBuilder);
     }
 
+    /**
+     * Perform acquire token silent call. If there is a valid access token in the cache, the sdk will return the access token; If
+     * no valid access token exists, the sdk will try to find a refresh token and use the refresh token to get a new access token. If refresh token does not exist
+     * or it fails the refresh, exception will be sent back via callback.
+     *
+     * @param scopes       The non-null array of scopes to be requested for the access token.
+     *                     MSAL always sends the scopes 'openid profile offline_access'.  Do not include any of these scopes in the scope parameter.
+     * @param account      {@link IAccount} represents the account to silently request tokens.
+     * @param authority    Optional. Can be passed to override the configured authority.
+     * @param forceRefresh True if the request is forced to refresh, false otherwise.
+     * @param callback     {@link AuthenticationCallback} that is used to send the result back. The success result will be
+     *                     sent back via {@link AuthenticationCallback#onSuccess(AuthenticationResult)}.
+     *                     Failure case will be sent back via {
+     * @link AuthenticationCallback#onError(MsalException)}.
+     */
+    public void acquireTokenSilentAsync(@NonNull final String[] scopes,
+                                        @NonNull final IAccount account,
+                                        @Nullable final String authority,
+                                        final boolean forceRefresh,
+                                        @NonNull final AuthenticationCallback callback) {
+        // TODO add support for forceRefresh
+        String requestAuthority = authority;
+
+        if (StringUtil.isEmpty(requestAuthority)) {
+            // TODO Do I use the List<Authority> from the config?
+            // Do I use the authority of the Account?
+            requestAuthority = "";
+        }
+
+        final MSALAcquireTokenSilentOperationParameters params = getSilentOperationParameters(
+                scopes,
+                requestAuthority
+        );
+
+        final MSALSilentTokenCommand silentTokenCommand = new MSALSilentTokenCommand(
```

Comment:
I suspect that this subclass is not needed, as `MSALTokenCommand` looks to basically cover the silent case